### ISSUE TITLE
Improve error reporting for syntax errors in rule files

### ISSUE THREAD

**Issue by pke** [eslint/eslint#5513]
**What version of ESLint are you using?**

2.3.0

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**

none

**What did you do? Please include the actual source code causing the issue.**

I wanted to include a custom rule

**What did you expect to happen?**

That the rule loads ok and triggers when a specific comment is found

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint crashed with a less helpful error message. It was really hard to track down, until I started to write a test and run it. Then I saw the real error.

See here http://stackoverflow.com/questions/35850285/custom-eslint-rule-throws-with-unexpected-reserved-word?noredirect=1#comment59383194_35850285

The ESLint error output of

```
SyntaxError: Failed to load plugin my-internal: Unexpected reserved word
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:413:25)
```

should be improved to show the position in the rules file where the error was raised.

**Comment by ilyavolodin** [comment]
I'm not sure if we can improve this, since the error is coming from the parser, and not from ESLint itself. But maybe we can have Espree provide a bit more information for the cases like that.

**Comment by nzakas** [comment]
We have no control over this. This isn't related to Espree, this error message is coming directly from Node.js because you are using invalid syntax. There's nothing we can do to improve the situation. If you want to pursue this, you'd need to file an issue with Node.js to see if they can give you better location information.
